src/Aspire.ProjectTemplates/templates/aspire-starter/AspireStarterApplication.1.Tests/AspireStarterApplication.1.Tests.csproj

```diff
@@ -1,3 +1,28 @@
+<!--#if (TestProjectFrameworkKind == "MSTest") -->
```

Comment:
The 2 projects are relatively different so it seems easier to have 2 configurations rather than many conditions.

Reply:
The other option is to completely separate them (different directories) and then conditionally exclude them from the template based on the selected framework. That avoids any in-file conditions like this at all.

Reply:
I wasn't sure how to avoid leaking the framework name in the project name. Whatever the framework I use, I expect the same project name (usually product.Tests or product.IntegrationTests...) which I couldn't figure with the conditional inclusion/exclusion.